### PR TITLE
Add groups to release notes

### PR DESCRIPTION
**Pull request by davidhewitt**
## Change Summary

Sets up five groups for release notes:
- Packaging - for dependency updates, python version changes etc
- New Features - hopefully obvious what goes here
- Changes - for changes to existing functionality (I include deprecations in here), and removals
- Performance - for performance change
- Fixes - for bugfixes

We can adjust these later if we want. These are the same groups I use in PyO3, and originally inspired by [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

Once merged, I will create these labels and backfill existing PRs.

Also added a CI check that at least one of these labels is added (or `relnotes-ignore`). @samuelcolvin we should make this a required check 😄 

## Related issue number

N/A

## Checklist

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @sydney-runkle

### CODE REVIEW

**Comment by davidhewitt** [comment]
A thought, do we want a `Performance` section? Seems like a hot topic for us recently.

**Comment by davidhewitt** [comment]
Sounded like generally was a yes, so I pushed it.

File: .github/release.yml [hunk 1]
```diff
@@ -2,3 +2,22 @@ changelog:
   exclude:
     labels:
       - ignore-for-release
+  categories:
+    - title: Packaging
+      labels:
+        - relnotes-packaging
+    - title: Added
```
Comment (samuelcolvin) [inline .github/release.yml:9]: New Feature?

**Review by lig — APPROVED** [review]
👍🏻 LGTM

**Pull request merged**